src/math/Vector3.js

```diff
@@ -683,9 +683,21 @@ Object.assign( Vector3.prototype, {
 
 	},
 
-	equals: function ( v ) {
+	equals: function ( v, epsilon ) {
 
-		return ( ( v.x === this.x ) && ( v.y === this.y ) && ( v.z === this.z ) );
+		if ( epsilon === undefined ) {
+
+			return ( ( v.x === this.x ) && ( v.y === this.y ) && ( v.z === this.z ) );
+
+		} else {
+
+			var x = Math.abs( this.x - v.x );
+			var y = Math.abs( this.y - v.y );
```

Comment:
There is no need to compute `y` if the test for `x` fails.

Reply:
So this pattern?
```javascript
var x = Math.abs( this.x - v.x );
if ( x > epsilon ) return false;

var y = Math.abs( this.y - v.y );
if ( y > epsilon ) return false;

return true;
```

Reply:
No. See #7346.

Reply:
Ah 😊 . I'll make a change...

Reply:
Love it like this!